### DISCUSSION TITLE
HASH BIF not defineable

### DISCUSSION THREAD

**Discussion by Peter-Jacob** [adesutherland/CREXX#342] [Q&A]
It is not possible to define a hash bif. The problem arises when you add HASH to **/...CREXX/lib/rxfns/rexx/CMakeLists.txt** and run a build, it stops immediately with:
```
"C:\Program Files\JetBrains\CLion 2020.2.4\bin\cmake\win\x64\bin\cmake.exe" --build C:\Users\PeterJ\CLionProjects\CREXX\cmake-build-debug --target all -j 10
ninja: error: 'lib/rxfns/rexx/hash.rexx', needed by 'lib/rxfns/rexx/_datei.rxbin', missing and no known rule to make it
```
this is not a big deal, just wonder why. I resolved the problem by naming the hash function fnv

**Comment by adesutherland** [comment]
That's weird! I will see if I can reproduce it ...

**Comment by rvjansen** [comment]
I don't have problems. I like the lowercase system info; only, I wonder if we don't need to return what uname() does, which is 'Darwin' on apple. Otoh, Java returns 'Mac OS X' so there we go.

I still need to stash a few parser artifacts which apparently are updated in the source tree. Peter, you may want to do that also.

best regards,

René.

**Reply by adesutherland** [reply]
I think you are talking about version - yes I went for lower case given modern aesthetics! I don't want to start a whole discussion elsewhere  but I agree the actual names for platforms should be standardised  - something like linux-i586, macosx-x64. Happy to follow the java convention but I can't find a list! Meantime I will change 'apple' to 'macosx'

**Comment by rvjansen** [comment]
oh, and I replied while not reading the actual message. Sorry!
